### DISCUSSION TITLE
date/time rollover does not delete archives

### DISCUSSION THREAD

**Discussion by john2home** [apache/logging-log4j2#2613] [Q&A]
hello 

in my log4j2.xml I have the following appender 

```
<RollingRandomAccessFile name="FILE_OLD" fileName=".log/log4j2-test-old.log" filePattern=".log/old/log4j2-test-old-%d{yyyy-MM-dd_HH-mm}.log.zip" append="true">
  <PatternLayout pattern="${pattern}" />
  <TimeBasedTriggeringPolicy interval="1"/>
  <DefaultRolloverStrategy max="3"/>
</RollingRandomAccessFile>
```
and  I have a test class to create log entries 
```
final var start = LocalDateTime.now();
var now = LocalDateTime.now();
do
{
  LOG.atWarn().log("log entry {}", now);
  Thread.sleep(10000l);
  now = LocalDateTime.now();
}
while (Duration.between(start, now).toMinutes() < 5);
```

i expected the zips older (or equal) than 3 minutes to be deleted based on the filePattern.  does this date/time rollover no longer work? 
 
in org.apache.logging.log4j.core.appender.rolling.AbstractRolloverStrategy.getEligibleFiles(String, String, String, boolean) 
````
        if (!PATTERN_COUNTER.matcher(logfilePattern).find()) {
            return eligibleFiles;
        }
````
the pattern will not match './log/old/log4j2-test-old-%d{yyyy-MM-dd_HH-mm}.log.zip' and the eligibleFiles is alwas empty.

log4j2 is 2.23.1, 
os is windows 10, 
java is openjdk 17

**Comment by rgoers** [comment]
The documentation on DefaultRolloverStrategy says ". If the date/time pattern is present it will be replaced with the current date and time values". In other words, the max attribute only limits the number of files within the same time period. To delete files the way you want to you will need to use a Delete action.
